contracts/strategies/traderjoe/TraderJoe.sol

```diff
@@ -84,62 +171,76 @@ contract TraderJoe is UUPSUpgradeable, StrategyOwnablePausableBaseUpgradeable {
         TraderJoeStorage storage strategyStorage = TraderJoeStorageLib
             .getStorage();
 
-        uint256 pairDepositTokenBalanceBeforeSwap = strategyStorage
-            .pairDepositToken
-            .balanceOf(address(this));
-
-        // swapping depositToken to pairDepositToken
+        // Swap half of depositToken to pairDepositToken.
         uint256 swapAmount = amount / 2;
         address[] memory path = new address[](2);
         path[0] = address(depositToken);
```

Comment:
So let's say the active bin now is 100, and we provide liquidity, that means now our contract provide liquidity for 50 bins: from 75-125. Some time after that the active bin is 200 and we provide liquidity again. Let's suppose that for now bins 75-125 only hold depositcurrency. We unnecessarily converted our half of our deposit currency in this case.

Reply:
It's stablecoin pool. And that's why there is a functionality that can adjust target bins. But I got what you mean and deposit and adjusting bin algorithm will be changed.

FYI, it's not fixed to target 51 bins. It could be 1 <= && <= 51.

Reply:
but do you adjust those target bins automatically? We cannot rely on management keep signing proposals to move bins.